spring/src/main/java/jet/spring/CustomSourceP.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package jet.spring;
+
+import com.hazelcast.jet.Source;
+import com.hazelcast.jet.Sources;
+import com.hazelcast.jet.Traverser;
+import com.hazelcast.jet.Traversers;
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.nio.Address;
+import com.hazelcast.spring.context.SpringAware;
+import jet.spring.dao.UserDao;
+import jet.spring.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+/**
+ * A processor which uses wired dao to find all users and emit them to downstream
+ */
+@SpringAware
+public class CustomSourceP extends AbstractProcessor {
+
+    private Traverser<User> traverser;
+
+    @Autowired
+    private transient UserDao userDao;
+
+    @Override
+    protected void init(@Nonnull Context context) {
+        traverser = Traversers.traverseIterable(userDao.findAll());
+    }
+
+    @Override
+    public boolean complete() {
+        return emitFromTraverser(traverser);
+    }
+
+    private static class MetaSupplier implements ProcessorMetaSupplier {
+
+        @Nonnull
+        @Override
+        public Function<Address, ProcessorSupplier> get(@Nonnull List<Address> addresses) {
+            return a -> c -> range(0, c).mapToObj(i -> new CustomSourceP()).collect(toList());
+        }
+
+        @Override
+        public int preferredLocalParallelism() {
+            return 1;
```

Comment:
you can use ProcessorMetaSupplier.dontParallize here and just have `ProcessorMetaSupplier.of(CustomSourceP::new)`